addons/mail/static/src/messaging/entity/entity_core.js

```diff
@@ -0,0 +1,246 @@
+odoo.define('mail.messaging.entity.core', function (require) {
+'use strict';
+
+const { patchClassMethods, patchInstanceMethods } = require('mail.messaging.utils');
+
+const registry = {};
+
+//------------------------------------------------------------------------------
+// Private
+//------------------------------------------------------------------------------
+
+/**
+ * @private
+ * @param {string} entityName
+ * @returns {Object}
+ */
+function _getEntryFromEntityName(entityName) {
+    if (!registry[entityName]) {
+        registry[entityName] = {
+            dependencies: [],
+            factory: undefined,
+            name: entityName,
+            patches: [],
+        };
+    }
+    return registry[entityName];
+}
+
+/**
+ * @private
+ * @param {string} entityName
+ * @param {string} patchName
+ * @param {Object} patch
+ * @param {Object} [param3={}]
+ * @param {Object} [param3.isOnClass=false]
+ */
+function _registerPatchEntity(entityName, patchName, patch, { isOnClass = false } = {}) {
```

Comment:
Not sure about the name of the method, and need to add doc explaining it + explain the params.